test/Base_Tests/src/Semantic/Error_Spec.enso

```diff
@@ -357,6 +357,35 @@ add_specs suite_builder =
             c2 . should_equal "finalizer"
             v2.to_vector . should_equal [1, 2]
 
+        group_builder.specify "should be caught in a loop" <|
+            fn_without_tco x =
+                if x > 10 then Panic.throw "fn_without_tco" else
+                    fn_without_tco x+1
+
+            p = Panic.catch Any handler=(.payload) <| fn_without_tco 0
+            p . should_equal "fn_without_tco"
+
+        group_builder.specify "should be caught in a TCO loop" <|
+            fn_with_tco x =
+                if x > 10 then Panic.throw "fn_with_tco" else
+                    @Tail_Call fn_with_tco x+1
+
+            p = Panic.catch Any handler=(.payload) <| fn_with_tco 0
+            p . should_equal "fn_with_tco"
+
+        # The same test as the one before, but it does not use the `<|` function.
+        group_builder.specify "should be caught in a TCO loop with a wrapper fn" <|
```

Comment:
As suggested by @JaroslavTulach, I have added this test. And as expected, this test succeeds, but the former ("should be caught in a TCO loop") fails. It seems that your suspicion, @JaroslavTulach, was right - this is probably caused by the fact that the `<|` function is inlineable and therefore there is no frame for it.

Reply:
EDIT: The test added in cce3c032e424eb6d6ddc85e4733a248afe087128 suggests otherwise. In that commit, I have introduced a test that doesn't use the `<|` method, but uses a method defined in a top scope, and it fails as well.